extensions/amp-a4a/0.1/streaming-utils.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Decodes readable stream from response and writes to given writeable stream.
+ * This function should be replaced with transform stream when well supported.
+ * @param {!Window} win
+ * @param {!Response} response
+ * @param {!WritableStreamDefaultWriter} writer
+ * @return {!Promise}
+ */
+export function streamResponseToWriter(win, response, writer) {
```

Comment:
where is this used? could you add it to this PR?

Reply:
sure, was trying to keep it simple but can make a larger pr.

Reply:
was wondering how this method is going to be used. you can still leave TODOs in the caller to make the PR not too large.

Reply:
Added the calling context.